### DISCUSSION TITLE
Location of installs via pip on Linux

### DISCUSSION THREAD

**Discussion by SAOJSM** [streamlink/streamlink#4775] [Q&A] [answered]
Before today, I used streamlink version 4.1.0.
But I found that afreeca.py not working. The path is /usr/lib/python3/dist-packages/streamlink/plugins/afreeca.py. I want to make contribution (fix it) to this PR. So I update streamlink to 4.3.0+21.g880ac1e2. After update. I can't find afreeca.py in that PATH. If anyone know where to find it. Please tell me. Thank you.

My system is ubuntu 20.02. And install using 
`pip3 install -U git+https://github.com/streamlink/streamlink.git`

**Comment by bastimeyer** [comment, marked as answer]
> `pip3 install -U git+https://github.com/streamlink/streamlink.git`

https://streamlink.github.io/install.html#pypi-package-and-source-code

> On Linux, when not using a virtual environment, it is recommended to install custom python packages like this only for the current user (see the --user parameter below), since system-wide packages can cause conflicts with the system's regular package manager.
> 
> **Those user-packages will be installed into ~/.local instead of /usr** and entry-scripts for running the programs can be found in ~/.local/bin, eg. ~/.local/bin/streamlink.

https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-user

**Reply by SAOJSM** [reply]
I know your means, but the problem not solve. If anyone know where to find it, please tell me. Thanks.

**Reply by bastimeyer** [reply]
> ubuntu 20.02
> The path is /usr/lib/python3/dist-packages/streamlink/

Uninstall your **system package**, because it's first of all broken according to what you've posted in #4774, and second, it's outdated. There's a reason why we don't recommend using the streamlink package on Ubuntu, because it inherits it from Debian from a version that's always outdated by a lot.

Then execute streamlink from `~/.local/bin/streamlink` instead of `/usr/bin/streamlink`. If you just type `streamlink`, then your command line shell will first need to resolve `streamlink` and will (most certainly) find `/usr/bin/streamlink` via the `PATH` environment variable.

So if your `PATH` env var doesn't contain `~/.local/bin`, then add it to it, perferrably at the first position, so that you're able to invoke your local streamlink install from pip by running just `streamlink`. If you're not sure how to do that, google it.
https://streamlink.github.io/install.html#pypi-package-and-source-code

Otherwise, install Streamlink in a virtual environment.
https://streamlink.github.io/install.html#virtual-environment

Good luck.

**Reply by SAOJSM** [reply]
But I check the newest version and reinstall. The version still streamlink 4.3.0+21.g880ac1e2. So I don't know why can't find the file I mentioned. And I already use find and locate. Nothing find.

**Reply by bastimeyer** [reply]
```
$ pip install --user -U git+https://github.com/streamlink/streamlink.git
...
Successfully installed streamlink-4.3.0+22.gb7b0353b

$ ls ~/.local/lib/python3.10/site-packages/streamlink/plugins/afreeca.py 
/home/basti/.local/lib/python3.10/site-packages/streamlink/plugins/afreeca.py
```

**Reply by SAOJSM** [reply]
Work, thank you.
